src/plugins/discover/public/utils/get_sharing_data.ts

```diff
@@ -107,7 +107,7 @@ export async function getSharingData(
 
         searchSource.setField('fields', fields);
       }
-      return searchSource.getSerializedFields(true);
+      return searchSource.getSerializedFields(true, false);
```

Comment:
@lukasolson should this be `false` only for ad-hoc data views?

Reply:
I'm not sure why you'd want to send the field list/mappings in the request since they don't seem to be necessary to the POST URL. Is there something that is being accessed from this list in the CSV generation?

Reply:
Reporting doesn't care about the parameters, as they just get forwarded to other dependencies on the server side when the report is generating:
* searchSource.getQuery
* tabify

If those dependencies do not need the field list or mappings, then I agree with you they don't seem to be necessary. It may have just been a carryover from an older implementation.

On a side note, I find the CSV export implementation that passes a searchsource object in the parameters to be overly complex, and since it doesn't reference a saved search it isn't user-friendly for advanced users that want to automate reports of any saved search. That's why I consider the `csv_searchsource` export type to be tech debt: https://github.com/elastic/kibana/issues/151190